cli/internal/fs/package_deps_hash.go

```diff
@@ -31,354 +21,352 @@ type PackageDepsOptions struct {
 	}
 
 	// Update the checked in hashes with the current repo status
-	gitStatusOutput, err := gitStatus(repoRoot.Join(p.PackagePath), p.InputPatterns)
+	gitStatusOutput, err := gitStatus(rootPath.Join(p.PackagePath), p.InputPatterns)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Could not get git hashes from git status")
 	}
-	currentlyChangedFiles := parseGitStatus(gitStatusOutput, p.PackagePath)
-	var filesToHash []string
-	for filename, changeType := range currentlyChangedFiles {
-		if changeType == "D" || (len(changeType) == 2 && string(changeType)[1] == []byte("D")[0]) {
-			delete(result, filename)
+
+	var filesToHash []turbopath.RelativeUnixPath
+	for filePath, status := range gitStatusOutput {
+		if status.x == "D" || status.y == "D" {
+			delete(result, filePath)
 		} else {
-			filesToHash = append(filesToHash, filepath.Join(p.PackagePath, filename))
+			filesToHash = append(filesToHash, filePath)
 		}
 	}
-	normalized := make(map[string]string)
-	// These paths are platform-dependent, but already relative to the package root
-	for platformSpecificPath, hash := range result {
-		platformIndependentPath := filepath.ToSlash(platformSpecificPath)
-		normalized[platformIndependentPath] = hash
-	}
-	hashes, err := GetHashableDeps(filesToHash, p.PackagePath)
+
+	convertedRootPath := turbopath.AbsoluteSystemPath(rootPath.ToString())
+	hashes, err := gitHashObject(convertedRootPath, filesToHash)
 	if err != nil {
 		return nil, err
 	}
-	for platformIndependentPath, hash := range hashes {
-		normalized[platformIndependentPath] = hash
+
+	// Zip up file paths and hashes together
+	for filePath, hash := range hashes {
+		result[filePath] = hash
 	}
-	return normalized, nil
+
+	return result, nil
 }
 
 // GetHashableDeps hashes the list of given files, then returns a map of normalized path to hash
 // this map is suitable for cross-platform caching.
-func GetHashableDeps(absolutePaths []string, relativeTo string) (map[string]string, error) {
-	fileHashes, err := gitHashForFiles(absolutePaths)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to hash files %v", strings.Join(absolutePaths, ", "))
-	}
-	result := make(map[string]string)
-	for filename, hash := range fileHashes {
-		// Normalize path as POSIX-style and relative to "relativeTo"
-		relativePath, err := filepath.Rel(relativeTo, filename)
+func GetHashableDeps(rootPath AbsolutePath, files []turbopath.AbsoluteSystemPath) (map[turbopath.RelativeUnixPath]string, error) {
+	output := make([]turbopath.RelativeUnixPath, len(files))
+	convertedRootPath := turbopath.AbsoluteSystemPath(rootPath.ToString())
+
+	for index, file := range files {
+		relativeSystemPath, err := file.Rel(convertedRootPath)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get relative path from %v to %v", relativeTo, relativePath)
+			return nil, err
 		}
-		key := filepath.ToSlash(relativePath)
-		result[key] = hash
+		output[index] = relativeSystemPath.ToRelativeUnixPath()
```

Comment:
And for bonus points: nope, we don't have a bug there because `doublestar` expects an `fs.FS` which is `/`-only. But it does want *unrooted* paths, which is just a fantastic troll, and probably a future type to add to `turbopath`.